cnd/src/ethereum/mod.rs

```diff
@@ -1,11 +1,78 @@
 #![warn(unused_extern_crates, missing_debug_implementations, rust_2018_idioms)]
 #![forbid(unsafe_code)]
 
-pub use web3::types::{
-    Address, Block, BlockId, BlockNumber, Bytes, Log, Transaction, TransactionReceipt,
-    TransactionRequest, H160, H2048, H256, U128, U256,
+pub use primitive_types::{
+    H160, H256, U128, U256
 };
 
+pub use ethbloom::{Bloom as H2048, Input};
+
+use fixed_hash::*;
+use impl_rlp::impl_fixed_hash_rlp;
+use impl_serde::impl_fixed_hash_serde;
+
+use impl_rlp::impl_uint_rlp;
+use impl_serde::impl_uint_serde;
+use uint::*;
+pub use uint::FromDecStrErr;
+
+use rustc_hex::{FromHex, ToHex};
+use serde::de::{Error, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt;
+
+pub type Index = U128;
+pub type Address = H160;
+
+construct_fixed_hash! {
+    pub struct H64(8);
```

Comment:
This is only used for the nonce if I see this correctly? We could also use a bigger one here and then get rid of the `fixed-hash` dependency.

Reply:
Tried and it doesn't pass tests, cant deserialise the test data. I would have to write a custom deserialiser?

Reply:
I think that would be preferable over taking another dependency. According to https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionreceipt, the field is either 1 or 0. You could write a custom deserializer that turns it into a boolean :)

Reply:
The H64 is used for nonce field in block. I have replaced the status field with U256. 
`pub nonce: Option<H64>`
Throws an error when trying to deserialise the H64 into a H256. I was thinking I could modify the deserialiser for H256 to accept H64 but afaik it is generated by macros

Reply:
actually i think ill just create our own H64 type, all it needs is deserialise/serialise